website/docs/en/config/devtool.mdx

```diff
@@ -4,54 +4,93 @@ import WebpackLicense from '@components/WebpackLicense';
 
 # Devtool
 
-The `devtool` configuration is used to control the behavior of the Source Map generation.
+Choose a style of source mapping to enhance the debugging process. These values can affect build and rebuild speed dramatically.
+
+Use the [SourceMapDevToolPlugin](/plugins/webpack/source-map-dev-tool-plugin) or [EvalSourceMapDevToolPlugin](/plugins/webpack/eval-source-map-dev-tool-plugin) for a more fine grained configuration.
 
 - **Type:**
 
 ```ts
-type Devtool =
-  | false
-  | 'eval'
-  | 'cheap-source-map'
-  | 'cheap-module-source-map'
-  | 'source-map'
-  | 'inline-cheap-source-map'
-  | 'inline-cheap-module-source-map'
-  | 'inline-source-map'
-  | 'inline-nosources-cheap-source-map'
-  | 'inline-nosources-cheap-module-source-map'
-  | 'inline-nosources-source-map'
-  | 'nosources-cheap-source-map'
-  | 'nosources-cheap-module-source-map'
-  | 'nosources-source-map'
-  | 'hidden-nosources-cheap-source-map'
-  | 'hidden-nosources-cheap-module-source-map'
-  | 'hidden-nosources-source-map'
-  | 'hidden-cheap-source-map'
-  | 'hidden-cheap-module-source-map'
-  | 'hidden-source-map'
-  | 'eval-cheap-source-map'
-  | 'eval-cheap-module-source-map'
-  | 'eval-source-map'
-  | 'eval-nosources-cheap-source-map'
-  | 'eval-nosources-cheap-module-source-map'
-  | 'eval-nosources-source-map';
+type Devtool = 'string' | false;
 ```
 
 - **Default:** `eval`
 
-The main types of Source Map generated behaviors are `source-map`, `eval`, `cheap`, `module`, `inline`, `nosources` and `hidden`, and they can be combined.
+## Configuration Guide
+
+### Step 1: Determine Debugging Needs
+
+- **Not required** → Set `devtool: false`
+  - Disables all debugging information
+  - Zero build overhead with maximum build speed
+- **Required** → Proceed to [Step 2](#step-2-define-debugging-requirements)
+
+### Step 2: Define Debugging Requirements
+
+- **Module-level positioning only** → Set `devtool: 'eval'`
```

Comment:
I feel that `eval` is no longer a recommended value. None of the frameworks based on webpack/Rspack use `eval` by default, and the code readability is quite poor.

Reply:
Even if we want to use `eval`, we should not use "devtool: 'eval'". We should write it like `process.env.NODE_ENV === 'production' ? 'source-map' : 'eval'`